### ISSUE TITLE
New build failures with Qt 5.14.2

### ISSUE THREAD

**Issue by felixonmars** [linuxdeepin/deepin-system-monitor#37]
```
src/gui/process_page_widget.cpp: In member function ‘void ProcessPageWidget::adjustStatusBarWidth()’:
src/gui/process_page_widget.cpp:420:58: warning: ‘const QRect QDesktopWidget::screenGeometry(int) const’ is deprecated: Use QGuiApplication::screens() [-Wdeprecated-declarations]
  420 |     QRect rect = QApplication::desktop()->screenGeometry();
      |                                                          ^
In file included from /usr/include/qt/QtWidgets/QDesktopWidget:1,
                 from src/gui/process_page_widget.cpp:35:
/usr/include/qt/QtWidgets/qdesktopwidget.h:79:67: note: declared here
   79 |     QT_DEPRECATED_X("Use QGuiApplication::screens()") const QRect screenGeometry(int screen = -1) const;
      |                                                                   ^~~~~~~~~~~~~~
src/gui/system_service_table_view.cpp: In member function ‘virtual bool SystemServiceTableView::eventFilter(QObject*, QEvent*)’:
src/gui/system_service_table_view.cpp:572:56: error: cannot dynamic_cast ‘event’ (of type ‘class QEvent*’) to type ‘class QKeyEvent*’ (target is not pointer or reference to complete type)
  572 |             auto *kev = dynamic_cast<QKeyEvent *>(event);
      |                                                        ^
g++ -c -pipe -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DQT_NO_DEBUG_OUTPUT -DQ_HOST_NAME=\"x86_64\" -DQ_HOST_X86_64 -DQ_OS_ARCH -DQ_OS_VERSION=\"\" -DQT_NO_DEBUG -DQT_MULTIMEDIAWIDGETS_LIB -DQT_WIDGETS_LIB -DQT_X11EXTRAS_LIB -DQT_MULTIMEDIA_LIB -DQT_GUI_LIB -DQT_DBUS_LIB -DQT_CONCURRENT_LIB -DQT_XML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I/build/deepin-system-monitor/src/deepin-system-monitor-5.6.1/nethogs/src/ -Isrc -isystem /usr/include/libdtk-5.2.0/DWidget -isystem /usr/include/libdtk-5.2.0/DCore -isystem /usr/include/libdtk-5.2.0/DGui -isystem /usr/include/libdtk-5.2.0/DWm -isystem /usr/include/qt -isystem /usr/include/qt/QtMultimediaWidgets -isystem /usr/include/qt/QtWidgets/5.14.2 -isystem /usr/include/qt/QtWidgets/5.14.2/QtWidgets -isystem /usr/include/qt/QtWidgets -isystem /usr/include/qt/QtX11Extras -isystem /usr/include/qt/QtMultimedia -isystem /usr/include/qt/QtGui/5.14.2 -isystem /usr/include/qt/QtGui/5.14.2/QtGui -isystem /usr/include/qt/QtGui -isystem /usr/include/qt/QtDBus -isystem /usr/include/qt/QtConcurrent -isystem /usr/include/qt/QtXml -isystem /usr/include/qt/QtNetwork -isystem /usr/include/qt/QtCore/5.14.2 -isystem /usr/include/qt/QtCore/5.14.2/QtCore -isystem /usr/include/qt/QtCore -Iout -I/usr/lib/qt/mkspecs/linux-g++ -o out/service_name_sub_input_dialog.o src/gui/service_name_sub_input_dialog.cpp
g++ -c -pipe -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DQT_NO_DEBUG_OUTPUT -DQ_HOST_NAME=\"x86_64\" -DQ_HOST_X86_64 -DQ_OS_ARCH -DQ_OS_VERSION=\"\" -DQT_NO_DEBUG -DQT_MULTIMEDIAWIDGETS_LIB -DQT_WIDGETS_LIB -DQT_X11EXTRAS_LIB -DQT_MULTIMEDIA_LIB -DQT_GUI_LIB -DQT_DBUS_LIB -DQT_CONCURRENT_LIB -DQT_XML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I/build/deepin-system-monitor/src/deepin-system-monitor-5.6.1/nethogs/src/ -Isrc -isystem /usr/include/libdtk-5.2.0/DWidget -isystem /usr/include/libdtk-5.2.0/DCore -isystem /usr/include/libdtk-5.2.0/DGui -isystem /usr/include/libdtk-5.2.0/DWm -isystem /usr/include/qt -isystem /usr/include/qt/QtMultimediaWidgets -isystem /usr/include/qt/QtWidgets/5.14.2 -isystem /usr/include/qt/QtWidgets/5.14.2/QtWidgets -isystem /usr/include/qt/QtWidgets -isystem /usr/include/qt/QtX11Extras -isystem /usr/include/qt/QtMultimedia -isystem /usr/include/qt/QtGui/5.14.2 -isystem /usr/include/qt/QtGui/5.14.2/QtGui -isystem /usr/include/qt/QtGui -isystem /usr/include/qt/QtDBus -isystem /usr/include/qt/QtConcurrent -isystem /usr/include/qt/QtXml -isystem /usr/include/qt/QtNetwork -isystem /usr/include/qt/QtCore/5.14.2 -isystem /usr/include/qt/QtCore/5.14.2/QtCore -isystem /usr/include/qt/QtCore -Iout -I/usr/lib/qt/mkspecs/linux-g++ -o out/process_table_view.o src/gui/process_table_view.cpp
make: *** [Makefile:1352: out/system_service_table_view.o] Error 1
make: *** Waiting for unfinished jobs....
src/gui/process_table_view.cpp: In member function ‘virtual bool ProcessTableView::eventFilter(QObject*, QEvent*)’:
src/gui/process_table_view.cpp:77:56: error: cannot dynamic_cast ‘event’ (of type ‘class QEvent*’) to type ‘class QKeyEvent*’ (target is not pointer or reference to complete type)
   77 |             auto *kev = dynamic_cast<QKeyEvent *>(event);
      |                                                        ^
make: *** [Makefile:1447: out/process_table_view.o] Error 1
```

**Comment by justforlxz** [comment]
```
diff --git a/deepin-system-monitor.pro b/deepin-system-monitor.pro
index 4b41229..c30c6e9 100644
--- a/deepin-system-monitor.pro
+++ b/deepin-system-monitor.pro
@@ -39,7 +39,6 @@ HEADERS += src/utils.h \
         src/compact_network_monitor.h \
         src/compact_disk_monitor.h \
         src/network_traffic_filter.h \
-        src/hashqstring.h \
         src/find_window_title.h \
         src/smooth_curve_generator.h \
         src/interactive_kill.h \
diff --git a/src/gui/process_table_view.cpp b/src/gui/process_table_view.cpp
index 3a18329..ae0f4b5 100644
--- a/src/gui/process_table_view.cpp
+++ b/src/gui/process_table_view.cpp
@@ -20,6 +20,7 @@
 #include <QMessageBox>
 #include <QProcess>
 #include <QTimer>
+#include <QKeyEvent>
 
 #include "kill_process_confirm_dialog.h"
 #include "main_window.h"
diff --git a/src/gui/system_service_table_view.cpp b/src/gui/system_service_table_view.cpp
index 15d18b4..d1b1f8c 100644
--- a/src/gui/system_service_table_view.cpp
+++ b/src/gui/system_service_table_view.cpp
@@ -10,6 +10,7 @@
 #include <QFutureWatcher>
 #include <QScrollBar>
 #include <QtConcurrent>
+#include <QKeyEvent>
 
 #include "common/error_context.h"
 #include "main_window.h"
diff --git a/src/hashqstring.h b/src/hashqstring.h
deleted file mode 100644
index 20522b3..0000000
--- a/src/hashqstring.h
+++ /dev/null
@@ -1,40 +0,0 @@
-/* -*- Mode: C++; indent-tabs-mode: nil; tab-width: 4 -*-
- * -*- coding: utf-8 -*-
- *
- * Copyright (C) 2011 ~ 2018 Deepin, Inc.
- *               2011 ~ 2018 Wang Yong
- *
- * Author:     Wang Yong <wangyong@deepin.com>
- * Maintainer: Wang Yong <wangyong@deepin.com>
- *
- * This program is free software: you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */ 
-
-#ifndef HASHQSTRING_H
-#define HASHQSTRING_H
-
-#include <QString>
-#include <string>
-
-namespace std {
-    template<> struct hash<QString>
-    {
-        size_t operator()(const QString& k) const
-            {
-                return hash<string>()(k.toStdString());
-            }
-    };
-}
-
-#endif // HASHQSTRING_H
diff --git a/src/process/stats_collector.cpp b/src/process/stats_collector.cpp
index eedc9c8..80bcc42 100644
--- a/src/process/stats_collector.cpp
+++ b/src/process/stats_collector.cpp
@@ -23,6 +23,7 @@
 #include <DDesktopEntry>
 #include <QPixmap>
 #include <QDateTime>
+#include <QUrl>
 
 #include "stats_collector.h"
 #include "process_entry.h"
diff --git a/src/utils.cpp b/src/utils.cpp
index 054025e..4d19745 100644
--- a/src/utils.cpp
+++ b/src/utils.cpp
@@ -49,7 +49,6 @@
 
 #include <X11/extensions/shape.h>
 
-#include "hashqstring.h"
 #include "utils.h"
 
 DCORE_USE_NAMESPACE
diff --git a/src/utils.h b/src/utils.h
index 747b445..7248711 100644
--- a/src/utils.h
+++ b/src/utils.h
@@ -32,7 +32,6 @@
 #include <QString>
 #include <unordered_set>
 #include "find_window_title.h"
-#include "hashqstring.h"
 #include <QDebug>
 
 #include <QVariant>
```

**Comment by felixonmars** [comment]
@justforlxz Verified to work fine. Thanks!

**Comment by z966768** [comment]
We have made some changes, you can try again with the latest version

**Comment by felixonmars** [comment]
@z966768 The latest version still didn't fix issues like in #42.
